test/components/views/rooms/wysiwyg_composer/EditWysiwygComposer-test.tsx

```diff
@@ -318,4 +342,290 @@ describe("EditWysiwygComposer", () => {
         await waitFor(() => expect(screen.getByRole("textbox")).toHaveTextContent(/🦫/));
         dis.unregister(dispatcherRef);
     });
+
+    describe("Keyboard navigation", () => {
+        const setup = async (
+            editorState = editorStateTransfer,
+            client = createTestClient(),
+            roomContext = defaultRoomContext,
+        ) => {
+            const spyDispatcher = jest.spyOn(defaultDispatcher, "dispatch");
+            customRender(false, editorState, client, roomContext);
+            await waitFor(() => expect(screen.getByRole("textbox")).toHaveAttribute("contentEditable", "true"));
+            return { textbox: screen.getByRole("textbox"), spyDispatcher };
+        };
+
+        beforeEach(() => {
+            mockPlatformPeg({ overrideBrowserShortcuts: jest.fn().mockReturnValue(false) });
+            jest.spyOn(EventUtils, "findEditableEvent").mockReturnValue(mockEvent);
+        });
+
+        function select(selection: SubSelection) {
+            return act(async () => {
+                await setSelection(selection);
+                // the event is not automatically fired by jest
+                document.dispatchEvent(new CustomEvent("selectionchange"));
+            });
+        }
+
+        describe("Moving up", () => {
+            it("Should not moving when caret is not at beginning of the text", async () => {
+                // When
+                const { textbox, spyDispatcher } = await setup();
+                const textNode = textbox.firstChild;
+                await select({
+                    anchorNode: textNode,
+                    anchorOffset: 1,
+                    focusNode: textNode,
+                    focusOffset: 2,
```

Comment:
I may be misreading this, but I think if we have both `anchorNode` and `focusNode` pointing at the same node, but different offsets, we don't have a caret in this test do we? I thought for a caret both offsets would have to be the same

Reply:
Here, there is a selection starting from the anchorOffset(0) to the focusOffset(1). The caret is on the focusOffset